spec/services/weather_man_spec.rb

```diff
@@ -36,64 +36,66 @@
+    let(:response_at_30) { { "history" => { "observations" => [obs_at_30] } } }
+    let(:response_at_45) { { "history" => { "observations" => [obs_at_45] } } }
+    let(:eight_am) { Time.zone.parse("Feb 20, 2015 at 8am") }
+    let(:two_pm) { Time.zone.parse("Feb 20, 2015 at 2pm") }
+    let(:eight_pm) { Time.zone.parse("Feb 20, 2015 at 8pm") }
+    let(:time) { Time.zone.parse("March 1, 2015 at 8am") }
+    let(:location) { 'knyc' }
+    let(:throttle) { 0 }
 
+    it "only pulls from the NOAA central park weather station", :vcr do
       expect(WeatherMan.api).to receive(:history_for).
-        with(time, "knyc").and_return(response)
-
-      WeatherMan.outdoor_temp_for(
-        time: time,
-        location: 10004,
-        throttle: 0
-      )
+        with(eight_am, "knyc").and_return(response_at_30)
+      WeatherMan.outdoor_temp_for(eight_am, location, throttle)
     end
 
     it "returns historical outdoor temperature", :vcr do
-      temperature = WeatherMan.outdoor_temp_for(
-        time: Time.zone.parse("Feb 20, 2015 at 8am"),
-        location: 10004,
-        throttle: 0
-      )
+      temperature = WeatherMan.outdoor_temp_for(eight_am, location, throttle)
       expect(temperature).to eq 4
-      temperature = WeatherMan.outdoor_temp_for(
-        time: Time.zone.parse("Feb 20, 2015 at 2pm"),
-        location: 10004,
-        throttle: 0
-      )
+      temperature = WeatherMan.outdoor_temp_for(two_pm, location, throttle)
       expect(temperature).to eq 19
-      temperature = WeatherMan.outdoor_temp_for(
-        time: Time.zone.parse("Feb 20, 2015 at 8pm"),
-        location: 10004,
-        throttle: 0
-      )
+      temperature = WeatherMan.outdoor_temp_for(eight_pm, location, throttle)
       expect(temperature).to eq 15
     end
+
+    it "caches responses" do
+      allow(WeatherMan.api).to receive(:history_for).and_return(response_at_30)
+      temp = WeatherMan.outdoor_temp_for(time, 'knyc', 0)
+      expect(temp).to eq 30
+
+      allow(WeatherMan.api).to receive(:history_for).and_return(response_at_45)
+      temp = WeatherMan.outdoor_temp_for(time, 'knyc', 0)
+      expect(temp).to eq 30
+    end
+
+    it "does not cache responses with no observations" do
+      allow(WeatherMan.api).to receive(:history_for).and_return(empty_response)
+      temperature = WeatherMan.outdoor_temp_for(time, 'knyc', 0)
+      expect(temperature).to eq nil
+
+      allow(WeatherMan.api).to receive(:history_for).and_return(response_at_45)
+      temperature = WeatherMan.outdoor_temp_for(time, 'knyc', 0)
+      expect(temperature).to eq 45
+    end
   end
 
-  describe ".fetch_wunderground_history" do
+  describe ".fetch_history_for" do
     it "returns different temps throughout the day", :vcr do
       time = Time.zone.parse("Feb 20, 2015 at 8am")
-      historical = WeatherMan.fetch_wunderground_history(time, 10004, 0)
+      response = WeatherMan.fetch_history_for(time, 'knyc', 0)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.